nixos/modules/services/misc/n8n.nix

```diff
@@ -19,47 +26,66 @@ in
       description = "Open ports in the firewall for the n8n web interface.";
     };
 
-    settings = lib.mkOption {
-      type = format.type;
-      default = { };
-      description = ''
-        Configuration for n8n, see <https://docs.n8n.io/hosting/environment-variables/configuration-methods/>
-        for supported values.
-      '';
-    };
-
-    webhookUrl = lib.mkOption {
-      type = lib.types.str;
-      default = "";
+    environment = lib.mkOption {
       description = ''
-        WEBHOOK_URL for n8n, in case we're running behind a reverse proxy.
-        This cannot be set through configuration and must reside in an environment variable.
+        Environment variables to pass to the n8n service.
+        See <https://docs.n8n.io/hosting/configuration/environment-variables/> for available options.
       '';
+      type = lib.types.submodule {
+        freeformType = with lib.types; attrsOf str;
+        options = {
+          GENERIC_TIMEZONE = lib.mkOption {
+            type = with lib.types; nullOr str;
```

Comment:
So an environment variable is simply not passed to systemd if it's set to `null`?
We should probably document this: https://search.nixos.org/options?channel=unstable&show=systemd.services.%3Cname%3E.environment

Reply:
Apparently yes 🤷🏽‍♂️